packages/viz/src/lib/sources/CARTOSource.ts

```diff
@@ -33,11 +33,18 @@ export class CARTOSource extends Source {
   private _type: 'sql' | 'dataset';
   // value it should be a dataset name or a SQL query
   private _value: string;
-  // Instantiation object
-  private _mapInstantiation: Promise<MapInstance>;
+
   // Internal credentials of the user
   private _credentials: Credentials;
 
+  private _layerProps?: CARTOLayerProps;
+
+  private _fieldStats?: FieldStats;
```

Comment:
We can assume not that we have just 1 field, but we'll probably change that in the future.

Reply:
Yes, it could be